apiary-metastore-metrics/src/main/java/com/expedia/apiary/extensions/metastore/metrics/CloudwatchReporter.java

```diff
@@ -0,0 +1,93 @@
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Gauge;
+
+import java.io.Closeable;
+import java.util.TimerTask;
+import java.util.List;
+import java.util.Map;
+
+import com.amazonaws.services.cloudwatch.AmazonCloudWatch;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchClientBuilder;
+import com.amazonaws.services.cloudwatch.model.MetricDatum;
+import com.amazonaws.services.cloudwatch.model.PutMetricDataRequest;
+import com.amazonaws.services.cloudwatch.model.PutMetricDataResult;
+import com.amazonaws.services.cloudwatch.model.StandardUnit;
+import com.amazonaws.services.cloudwatch.model.Dimension;
+
+/**
+ * TODO: need to update this to implement CodahaleReporter when migrating to hive 3.0.0, 
+ * similar to https://github.com/apache/hive/blob/master/common/src/java/org/apache/hadoop/hive/common/metrics/metrics2/JsonFileMetricsReporter.java
+ */
+
+class CloudwatchReporter implements Closeable {
+    private java.util.Timer timer = null;
+    private AmazonCloudWatch cw = null;
+    private MetricRegistry metricRegistry;
+
+    public CloudwatchReporter(MetricRegistry metricRegistry) {
+        this.metricRegistry=metricRegistry;
+    }
+
+    public void start() {
+      this.cw = AmazonCloudWatchClientBuilder.defaultClient();
+      this.timer = new java.util.Timer(true);
+
+      timer.schedule(new TimerTask() {
+        @Override
+        public void run() {
+            String namespace = System.getenv("CLOUDWATCH_NAMESPACE");
+            Dimension dimension = new Dimension().withName("ecsTaskId") .withValue(System.getenv("ECS_TASK_ID"));
+            //all interested metrics are gauges so only writing those to cloudwatch
+            for (Map.Entry<String,Gauge> entry : metricRegistry.getGauges().entrySet()) {
+                try
+                {
+                    String metricName = entry.getKey();
+                    Double metricValue;
+                    if(entry.getValue().getValue() instanceof Long)
+                    {
+                        metricValue = ((Long)(entry.getValue().getValue())).doubleValue();
+                    }
+                    else if(entry.getValue().getValue() instanceof Integer)
+                    {
+                        metricValue = ((Integer)(entry.getValue().getValue())).doubleValue();
+                    }
+                    else
+                    {
+                        metricValue = (double)entry.getValue().getValue();
+                    }
+                    MetricDatum datum = new MetricDatum().withMetricName(metricName).withUnit(StandardUnit.None).withValue(metricValue).withDimensions(dimension);
+                    PutMetricDataRequest request = new PutMetricDataRequest().withNamespace(namespace) .withMetricData(datum);
+                    PutMetricDataResult response = cw.putMetricData(request);
+                } catch (Exception e) {
+                    //ignore threads.deadlocks collection
```

Comment:
What does this mean? Shouldn't we at least log something at info/warn level here?